### PR TITLE
[flang][driver] add -flang-deprecated-no-hlfir hidden option

### PR DESCRIPTION
**Pull request by jeanPerier**
Patch 1/3 of the transition to use the HLFIR step by default in lowering as described in https://discourse.llvm.org/t/rfc-enabling-the-hlfir-lowering-by-default/72778/7

This option will allow to lower code without the HLFIR step during a grace period as described in the RFC. It is not meant to be a long term switch for flang.

### CODE REVIEW

File: flang/lib/Frontend/CompilerInvocation.cpp [hunk 1]
```diff
@@ -1073,6 +1073,12 @@ bool CompilerInvocation::createFromArgs(
     res.loweringOpts.setLowerToHighLevelFIR(true);
   }
 
+  // -flang-deprecated-no-hlfir
+  if (args.hasArg(clang::driver::options::OPT_flang_deprecated_no_hlfir) &&
+      !args.hasArg(clang::driver::options::OPT_emit_hlfir)) {
```
Comment (tblah) [inline flang/lib/Frontend/CompilerInvocation.cpp:1078]: nit: specifying -flang-depricated-nohlfir and -flang-experimental-hlfir together should probably be an error or at least a warning

**Review by tblah — CHANGES_REQUESTED** [review]
Thanks for this!

Please could you also add testing in flang/Driver/frontend-forwarding.f90, flang/test/HLFIR/hlfir-flang.f90

**Comment by jeanPerier** [comment]
> Please could you also add testing in flang/Driver/frontend-forwarding.f90, flang/test/HLFIR/hlfir-flang.f90

Thanks for the review @tblah. I could not find flang/test/HLFIR/hlfir-flang.f90 test, did you mean flang/test/HLFIR/hlfir-**flags**.f90 (in which case it is updated in this PR already).

I added an error when -flang-deprecated-no-hlfir and -flang-experimental-hlfir are both specified and updated frontend-forwarding.f90.

**Review by vzakhari — APPROVED** [review]
Thank you, Jean!

**Review by tblah — APPROVED** [review]
> Thanks for the review @tblah. I could not find flang/test/HLFIR/hlfir-flang.f90 test, did you mean flang/test/HLFIR/hlfir-**flags**.f90 (in which case it is updated in this PR already).

Yes I meant hlfir-flags.f90. Sorry about that. Thanks for the changes!

**Comment by jeanPerier** [comment]
> Yes I meant hlfir-flags.f90. Sorry about that. Thanks for the changes!
No problem, thanks for the review and comments!

**Pull request merged**